woocommerce/wc-admin
# Try: Register plugins under separate navigation menu

Update the nav to the "Plugins" approach.

Fixes https://github.com/woocommerce/woocommerce-admin/issues/5530

* Locks core menu items, creating dependency on WCA features within the navigation.
* Adds methods to register plugin categories and items.
* Adds extensions group to client.
* Updates the example plugin.

Note: this PR does not update the styling to the group title "Extensions."  The group title styling is different than that shown in https://github.com/woocommerce/woocommerce-admin/issues/5530.  I think this is something we may want to tackle upstream.

### Screenshots
<img width="301" alt="Screen Shot 2020-11-04 at 10 45 41 AM" src="https://user-images.githubusercontent.com/10561050/98134294-98a2e880-1e8c-11eb-8fbe-c32867ac9a83.png">


### Detailed test instructions:

1. Run the extension `npm run example -- --ext=add-navigation-items` and activate.
1. Note the extensions group and items/categories.
1. Make sure you can't register items in the main menu.
1. Make sure all relevant nav items still exist in the core nav menu.
